src/components/Match/BuildingMap/BuildingMap.jsx

```diff
@@ -22,34 +37,171 @@ export default function BuildingMap({
     // building data in correct order
     // determine ancient display by match winner
     for (let i = 0; i < bits.length; i += 1) {
-      const type = buildingData[i].id.slice(0, 1) === 't' ? 'tower' : 'racks';
-      const side = buildingData[i].id.slice(-1) === 'r' ? '_radiant.png' : '_dire.png';
-      const d = {
+      let type = buildingData[i].id.slice(0, 1);
+      type =
+        (type === 't' && 'tower') ||
+        (type === 'b' && (buildingData[i].id.slice(1, 2) === 'm' ? 'melee_rax' : 'range_rax')) ||
+        (type === 'a' && 'fort');
+      const side = buildingData[i].id.slice(-1) === 'r' ? 'good' : 'bad';
+      const tier = Number(buildingData[i].id.slice(1, 2)) || '';
+      let lane = buildingData[i].id.slice(2, 3);
+      lane = (tier !== 4 && (
+        (lane === 't' && 'top') ||
+        (lane === 'm' && 'mid') ||
+        (lane === 'b' && 'bot') || '')) || '';
+
+      const key = `npc_dota_${side}guys_${type}${tier}${lane && `_${lane}`}`;
+      const title =
+        strings[`${type.includes('rax') ? 'building_' : 'objective_'}${type}${tier}${type.includes('rax') ? '' : lane && `_${lane}`}`];
+
+      const destroyedBy = match.version && match.players
```

Comment:
check for the field you're using instead of match.version.  Probably really old versions don't have killed field.